### PR TITLE
!!! FEATURE: Cleanup Subgraph API

### PR DESCRIPTION
**Pull request by bwaidelich**
Unifies naming and signatures of the `ContentSubgraphInterface`:

* `findSucceedingSiblings(NodeAggregateId $sibling, ...)` => `findSucceedingSiblingNodes(NodeAggregateId $siblingNodeAggregateId, ...)`
* `findPrecedingSiblings(NodeAggregateId $sibling, ...)` => `findPrecedingSiblingNodes(NodeAggregateId $siblingNodeAggregateId, ...)`
* `findDescendants(NodeAggregateIds $entryNodeAggregateIds, ...)` => `findDescendantNodes(NodeAggregateId $entryNodeAggregateId, ...)`
* `findReferencedNodes(...)` => `findReferences(...)`
* `findReferencingNodes(...)` => `findBackReferences(...)`
* `findNodePath(...): ?NodePath` => `retrieveNodePath(...): NodePath`

Related: #4075

### CODE REVIEW

**Comment by bwaidelich** [comment]
**Note:** Based on #4076

**Comment by bwaidelich** [comment]
anyone knows why most of the checks are missing (behat tests, ...)?

**Comment by ahaeslich** [comment]
> anyone knows why most of the checks are missing (behat tests, ...)?

I think this is happening because your are targeting another feature branch. Have a look at .github/workflows/build.yml

File: Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphInterface.php [hunk 1]
```diff
@@ -57,107 +47,94 @@
...
-        NodeAggregateId $sibling,
-        Filter\FindSucceedingSiblingsFilter $filter
-    ): Nodes;
+    public function findChildNodeConnectedThroughEdgeName(NodeAggregateId $parentNodeAggregateId, NodeName $edgeName): ?Node;
 
     /**
-     * @param NodeAggregateId $sibling
-     * @param FindPrecedingSiblingsFilter $filter
-     * @return Nodes
+     *
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphInterface.php]: "recursively find all nodes underneath the entry node aggregate ID, which match the specified filter and return them as a flat list".

File: Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphInterface.php [hunk 2]
```diff
@@ -57,107 +47,94 @@
...
-    public function findPrecedingSiblings(
-        NodeAggregateId $sibling,
-        Filter\FindPrecedingSiblingsFilter $filter,
-    ): Nodes;
+    public function findDescendantNodes(NodeAggregateId $entryNodeAggregateId, Filter\FindDescendantNodesFilter $filter): Nodes;
 
     /**
-     * @param NodeAggregateId $nodeAggregateId
-     * @return NodePath
+     * Recursively find all nodes underneath the $entryNodeAggregateId that match the specified $filter and return them as a tree
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphInterface.php:93]: recursively find nodes underneath entrynodeagg, keeping the tree structure intact.

This returns a fragment of the existing tree structure.

**Pull request merged**